datalad/core/distributed/push.py

```diff
@@ -498,7 +499,13 @@ def _push(dspath, content, target, data, force, jobs, res_kwargs, pbars,
     depvar = 'remote.{}.datalad-publish-depends'.format(target)
     # list of remotes that are publication dependencies for the
     # target remote
-    publish_depends = ensure_list(ds.config.get(depvar, []))
+    # multiple dependencies could come from multiple declarations
+    # of such a config items, but each declaration could also
+    # contain a white-space separated list of remote names
+    # see https://github.com/datalad/datalad/issues/6867
```

Comment:
where do we document/allow for a white-space separated list of names?

Rationale for asking:  Indeed ATM git does not allow white-space in the name of the remote, and it seems we do not "disallowed" it explicitly since just allow for `git` to error out:

```shell
$> datalad siblings add -s "my name" --url https://example.com
CommandError: 'git -c diff.ignoreSubmodules=none remote add 'my name' https://example.com' failed with exitcode 128
fatal: 'my name' is not a valid remote name
```

so if in the future `git` for some reason decides to support that (unlikely, but not impossible IMHO) such splitting by space might become buggy.  So I wondered to ask where we already document/allow for that, and may be we should add some explicit checking for the name to not contain spaces (since could lead to fail of this added logic)

Reply:
Hmm, I cannot find a user-facing documentation that made me conclude that we allow(ed) for this. Maybe it was a comment, but I cannot find that either.

Reply:
then since we didn't have that documented, and looking at original code above and behavior of ensure_list:

```shell
$> python -c 'from datalad.utils import ensure_list; print(ensure_list("w1 w2"))'
['w1 w2']
```
that it doesn't split by space, I think we never had such a feature -- should we really add it along with this?  may be just a matter of adding `get_all=True` and be done?